internal/iam/query.go

```diff
@@ -176,4 +176,41 @@ const (
 	)
 	select role_id as role_id, role_scope as scope_id, role_grant as grant from final;
 		`
+
+	listRolesTemplate = `
+	select *
+		from iam_role
+		where %s
+	order by create_time desc, public_id asc
+		limit %d;
+	`
```

Comment:
With this simple a query, can we not just use `db.reader.SearchWhere` and `db.WithOrder` and `db.WithLimit`? That means we can skip these queries altogether.